plotting/basemap.py

```diff
@@ -1,120 +1,98 @@
+    filename = _get_filename(plot_proj.proj4_params["proj"], extent)
+    filename = "".join([CACHE_DIR, "/", filename])
 
-    def get_resolution(h: float, w: float) -> str:
-        area_km = (h * w) / (1000 * 1000)
+    land_shp = _get_shapefile(plot_resolution)
 
-        if area_km < 10000:
-            return "f"  # full resolution
-        elif area_km < 100000:
-            return "h"  # high resolution
-        elif area_km < 1000000:
-            return "i"  # intermediate resolution
-        elif area_km < 10000000:
-            return "l"  # low resolution
-        else:
-            return "c"  # crude resolution
-
-    if _maps_cache.get(filename) is None or True:
-        filename = "".join([CACHE_DIR, "/", filename])
+    if not os.path.exists(filename):
 
         try:
-            basemap = pickle.load(open(filename))
+            fig, map_plot = pickle.load(open(filename))
         except:
-            if projection in ["npstere", "spstere"]:
-                basemap = Basemap(
-                    resolution=get_resolution(height, width),
-                    area_thresh=((height * width) / (1000 * 1000))
-                    * 0.00001,  # display islands whose area is 0.001% of displayed area
-                    ellps="WGS84",
-                    projection=projection,
-                    boundinglat=min_lat,
-                    lon_0=center[1],
-                )
-            elif projection == "merc":
-                basemap = Basemap(
-                    resolution="c",
-                    # area_thresh=((height*width)/(1000*1000))*0.00001 , #display islands whose area is 0.001% of displayed area
-                    ellps="WGS84",
-                    projection=projection,
-                    llcrnrlat=height[0],
-                    llcrnrlon=width[0],
-                    urcrnrlat=height[1],
-                    urcrnrlon=width[1],
-                )
-            else:
-                basemap = Basemap(
-                    resolution=get_resolution(height, width),
-                    area_thresh=((height * width) / (1000 * 1000))
-                    * 0.00001,  # display islands whose area is 0.001% of displayed area
-                    ellps="WGS84",
-                    projection=projection,
-                    lat_0=center[0],
-                    lon_0=_convert_to_bounded_lon(center[1]),
-                    height=height,
-                    width=width,
-                )
-            basemap.filename = filename
-
-            def do_pickle(basemap: Basemap, filename: str) -> None:
-                pickle.dump(basemap, open(filename, "wb"), -1)
+            fig = plt.figure(figsize=figuresize, dpi=dpi)
+
+            map_plot = plt.axes(projection=plot_proj, facecolor="dimgrey")
+            map_plot.set_extent(extent, crs=plot_proj)
+
+            map_plot.add_geometries(
+                land_shp.geometries(),
+                crs=ccrs.PlateCarree(),
```

Comment:
can we cache this object between requests? Apparently this guy is stupid slow to instantiate with LCC.

Reply:
I was trying to achieve something similar by pickling the plot after setting the extent and adding the land geometries. I'm not sure if you could effectively cache the `ccrs.LambertConformal` object since you have to specify the central lat/lon when you create it, otherwise the area plotted may be skewed depending on where it is relative to the default central lat/lon (39,-96).